Library/Homebrew/cxxstdlib.rb

```diff
@@ -16,20 +16,16 @@ def self.create(type, compiler)
     if type && ![:libstdcxx, :libcxx].include?(type)
       raise ArgumentError, "Invalid C++ stdlib type: #{type}"
     end
-    klass = GNU_GCC_REGEXP === compiler.to_s ? GnuStdlib : AppleStdlib
-    klass.new(type, compiler)
+    new(type, compiler)
   end
 
   def self.check_compatibility(formula, deps, keg, compiler)
     return if formula.skip_cxxstdlib_check?
 
     stdlib = create(keg.detect_cxx_stdlibs.first, compiler)
-
-    begin
-      stdlib.check_dependencies(formula, deps)
-    rescue CompatibilityError => e
-      opoo e.message
-    end
+    stdlib.check_dependencies(formula, deps)
+  rescue CompatibilityError
+    opoo $!.message
```

Comment:
Why change the error handling? The new version is less correct (the scope of the rescue has changed). What is the point of switching from `=> e` to `$?`?).